core/arch/arm/mm/core_mmu.c

```diff
@@ -200,107 +200,155 @@ static struct tee_mmap_region *find_map_by_pa(unsigned long pa)
 	return NULL;
 }
 
+static bool pbuf_is_special_mem(paddr_t pbuf, size_t len,
+				const struct core_mmu_phys_mem *start,
+				const struct core_mmu_phys_mem *end)
+{
+	const struct core_mmu_phys_mem *mem;
+
+	for (mem = start; mem < end; mem++) {
+		if (core_is_buffer_inside(pbuf, len, mem->addr, mem->size))
+			return true;
+	}
+
+	return false;
+}
+
+extern const struct core_mmu_phys_mem __start_phys_nsec_ddr_section;
+extern const struct core_mmu_phys_mem __end_phys_nsec_ddr_section;
+
+static bool pbuf_is_nsec_ddr(paddr_t pbuf, size_t len)
+{
+	return pbuf_is_special_mem(pbuf, len, &__start_phys_nsec_ddr_section,
+				    &__end_phys_nsec_ddr_section);
+}
+
+bool core_mmu_nsec_ddr_is_defined(void)
+{
+	return &__start_phys_nsec_ddr_section != &__end_phys_nsec_ddr_section;
+}
+
+#define MSG_MEM_INSTERSECT(pa1, sz1, pa2, sz2) \
+	EMSG("[%" PRIxPA " %" PRIxPA "] intersecs [%" PRIxPA " %" PRIxPA "]", \
+			pa1, pa1 + sz1, pa2, pa2 + sz2)
+
 #ifdef CFG_SECURE_DATA_PATH
 extern const struct core_mmu_phys_mem __start_phys_sdp_mem_section;
 extern const struct core_mmu_phys_mem __end_phys_sdp_mem_section;
 
 static bool pbuf_is_sdp_mem(paddr_t pbuf, size_t len)
 {
+	return pbuf_is_special_mem(pbuf, len, &__start_phys_sdp_mem_section,
+				    &__end_phys_sdp_mem_section);
+}
+
+struct mobj **core_sdp_mem_create_mobjs(void)
+{
 	const struct core_mmu_phys_mem *mem;
+	struct mobj **mobj_base;
+	struct mobj **mobj;
+	int cnt = &__end_phys_sdp_mem_section - &__start_phys_sdp_mem_section;
 
-	for (mem = &__start_phys_sdp_mem_section;
-	     mem < &__end_phys_sdp_mem_section; mem++)
-		if (core_is_buffer_inside(pbuf, len, mem->addr, mem->size))
-			return true;
+	/* SDP mobjs table must end with a NULL entry */
+	mobj_base = calloc(cnt + 1, sizeof(struct mobj *));
+	if (!mobj_base)
+		panic("Out of memory");
+
+	for (mem = &__start_phys_sdp_mem_section, mobj = mobj_base;
+	     mem < &__end_phys_sdp_mem_section; mem++, mobj++) {
+		*mobj = mobj_phys_alloc(mem->addr, mem->size,
+					TEE_MATTR_CACHE_CACHED,
+					CORE_MEM_SDP_MEM);
+		if (!*mobj)
+			panic("can't create SDP physical memory object");
+	}
+	return mobj_base;
+}
+
+static void check_sdp_intersection_with_nsec_ddr(void)
+{
+	const struct core_mmu_phys_mem *sdp_start =
+		&__start_phys_sdp_mem_section;
+	const struct core_mmu_phys_mem *sdp_end = &__end_phys_sdp_mem_section;
+	const struct core_mmu_phys_mem *ddr_start =
+		&__start_phys_nsec_ddr_section;
+	const struct core_mmu_phys_mem *ddr_end = &__end_phys_nsec_ddr_section;
+	const struct core_mmu_phys_mem *sdp;
+	const struct core_mmu_phys_mem *nsec_ddr;
+
+	if (sdp_start == sdp_end || ddr_start == ddr_end)
```

Comment:
minor: test on `ddr_start == ddr_end` is sufficient to prevent a useless scan. Could even remove the full test.